### PR TITLE
Admin Menu: Remove style dependencies

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/wp-calypso/issues/51535.
Counterpart of 659-gh-Automattic/wpcomsh (hotfix for Atomic sites).

#### Changes proposed in this Pull Request:
Removes the masterbar styles dependencies from the admin menu styles. They are unnecessary, and was causing some nav unification issues when they were missing (like when [AMP is enabled](https://github.com/Automattic/jetpack/blob/f8e877deb5f5b74f3e99bfabdd2603c683231998/projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php#L138-L144)).

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_Simple sites_
* Apply the diff below to your WP.com sandbox.
* Sandbox a WP.com simple site.
* Visit a WP Admin of the sandboxed site.
* Make sure there are no visual regressions in the unified menu.

_Atomic sites_
* Install and activate the Jetpack Beta plugin.
* Go to Jetpack > Jetpack Beta, and switch to the branch of this PR.
* Install and activate the AMP plugin.
* Go to AMP > Settings and activate the "Standard" mode.
* Make sure there are no visual regressions in the unified menu.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php [hunk 1]
```diff
@@ -198,14 +198,6 @@ public function add_admin_menu_separator( $position, $cap = 'read' ) {
 	public function enqueue_scripts() {
 		$is_wpcom = defined( 'IS_WPCOM' ) && IS_WPCOM;
 
-		if ( $is_wpcom ) {
-			$style_dependencies = array( 'wpcom-admin-bar', 'wpcom-masterbar-css' );
-		} elseif ( $this->is_rtl() ) {
-			$style_dependencies = array( 'a8c-wpcom-masterbar-rtl', 'a8c-wpcom-masterbar-overrides-rtl' );
-		} else {
-			$style_dependencies = array( 'a8c-wpcom-masterbar', 'a8c-wpcom-masterbar-overrides' );
-		}
```
[the remark this answers is not in the document]
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php:207]: I cannot reproduce that 🤔 I noted that you used a different site for the after screenshot. Do you get the same behavior if you test on the same Atomic site you used for the before screenshot?

This is what I get (note how Jetpack Beta is running the branch of this PR):

Screen Shot 2021-03-31 at 14 46 41

[the remark this answers is not in the document]
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php:207]: Never mind, I didn't have AMP enabled. I can reproduce it now

[the remark this answers is not in the document]
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php:207]: Should've been fixed in 1406f5f.

[the remark this answers is not in the document]
Comment (cpapazoglou) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php:207]: I was about to suggest the same solution. Let me now check both Atomic & Simple!

**Review by cpapazoglou — APPROVED** [review]
While I can't understand it now, removing this dependencies fixes the problem with the disabled masterbar and I can't spot any regression both in Atomic & Simple.

LGTM!

**Comment by sdixon194** [comment]
Tested on simple and atomic and both work for me 👍

**Review by sdixon194 — APPROVED** [review]

**Comment by mmtr** [comment]
Deployed to WP.com in r223675-wpcom.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.6` in 372bb7bfbe6da27e12f128af89abf0a4f7294561

**Pull request merged**